enumerable.rb

```diff
@@ -0,0 +1,163 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+
+module Enumerable
+  # 1 my_each
+  def my_each
+    to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a[i]
+      i += 1
+    end
+  end
+
+  # 2 my_each_with_index
+  def my_each_with_index
+    to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield(to_a[i], i)
+      i += 1
+    end
+  end
```

Comment:
#my_each_with_index returns an enumerator if no block is given
#my_each_with_index when a block is given when self is an array returns the array itself after calling the given block once for each element in self
#my_each_with_index when a block is given when self is a range returns the range itself after calling the given block once for each element in self